### PR TITLE
🔧 update hotspots script

### PR DESCRIPTION
**Pull request by dmiller15**
## Description

Updates the hotspot scripts to handle newly defined inputs.

Closes https://github.com/d3b-center/bixu-tracker/issues/952

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Ran locally using example files from https://github.com/d3b-center/bixu-tracker/issues/952 and this VCF https://cavatica.sbgenomics.com/u/zhangb1/kf-somatic-tools-test/files/6047f398252bc31f2a8c8678/

**Test Configuration**:
* Environment: docker pull dmiller15/hotspots:0.1.0
* Test files:
   - Hotspot examples here: https://github.com/d3b-center/bixu-tracker/issues/952
   - VCF: https://cavatica.sbgenomics.com/u/zhangb1/kf-somatic-tools-test/files/6047f398252bc31f2a8c8678/

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have committed any related changes to the PR

### CODE REVIEW

**Comment by dmiller15** [comment]
I've added a Docker that you guys can use to test. See the `Environment` section of the PR.

File: scripts/hotspot.py [hunk 1]
```diff
@@ -58,39 +85,75 @@ def csq_sanity(field,value):
...
+        hs_dict (dict): dict of key: protein name and values: unique HGVSp short names
     Return:
-        list: returns a list of unique hotspot values from the file
+        list: returns a dict of key: protein name and values: unique HGVSp short names
     """
-    hs_list = []
-    with open(infile) as fh:
+    if hs_dict is None:
+        hs_dict = {}
+    opener = gzip.open if infile.endswith('.gz') else open
```
Comment (migbro) [inline scripts/hotspot.py:125]: Ohhh, fancy! 🎩🧐

**Review by NHJohnson — APPROVED** [review]
I was able to run it in the docker with BS_M5FM63EB_TEST.mutect2_somatic.gatk.soft_filtered.vcf.gz and cancerhotspots.v2.maf.gz (protein hotspots) and it worked! 2844 hotspots annotated.

**Review by migbro — CHANGES_REQUESTED** [review]
Only request I have is to have pysam indeo the vcf after creation

File: scripts/hotspot.py [hunk 2]
```diff
@@ -58,39 +85,75 @@ def csq_sanity(field,value):
...
+def process_protein_and_hgvsp(infile,protein_colname="Hugo_Symbol",hgvsp_colname="HGVSp_Short",hs_dict=None):
+    """Function to parse a column-name-labeled, tab-delimited file and return protein name and HGVSp
+    Args:
+        infile (file): Tab-delimited MAF file containing hotspot information
+        protein_colname (string): Name of the column that contains the Hugo Symbol values
+        hgvsp_colname (string): Name of the column that contains the HGVSp Short values
+        hs_dict (dict): dict of key: protein name and values: unique HGVSp short names
     Return:
-        list: returns a list of unique hotspot values from the file
+        list: returns a dict of key: protein name and values: unique HGVSp short names
```
[the remark this answers is not in the document]
Comment (dmiller15) [reply scripts/hotspot.py:121]: True. Adjusting

File: scripts/hotspot.py [hunk 3]
```diff
@@ -58,39 +85,75 @@ def csq_sanity(field,value):
...
-    unique_hs = list(set([items for sublist in hs_list for items in sublist]))
-    print(csq_sanity(field,unique_hs[0]))
-    return unique_hs
-
-def process_vcf(infile,outfile,hslist,fid,aa='Amino_acids',pos='Protein_position',allele='ALLELE_NUM'):
-    """Function to take the input VCF and add INFO annotations for the ALT alleles that cause protein changes at hotspots
-
+            hugo = row[header.index(protein_colname)]
+            hgvsp = row[header.index(hgvsp_colname)]  #('A2ML1', 'p.V598=')
+            if hugo not in hs_dict:
```
[the remark this answers is not in the document]
Comment (dmiller15) [reply scripts/hotspot.py:139]: I'll look into that.

File: scripts/hotspot.py [hunk 4]
```diff
@@ -101,54 +164,66 @@ def process_vcf(infile,outfile,hslist,fid,aa='Amino_acids',pos='Protein_position
...
-#                hotspot_alleles.append(i.split('|')[csq_items.index(allele))
-#        hotspot_allele_nums = set(hotspot_alleles)
-        hotspot_array = [0] * len(rec.alts)
-        for i in hotspot_allele_nums:
-            total_hotspots_found += 1
-            hotspot_array[int(i)-1] = 1
+        genomic_hotspot_found = False
+        if rec.chrom in genomic_hs:
+            for region in genomic_hs[rec.chrom]:
+                if rec.start == int(region[0]) and rec.stop == int(region[1]): # we're only marking it if the rec start and stop match exactly to the genomic regions
```
[the remark this answers is not in the document]
Comment (dmiller15) [reply scripts/hotspot.py:170]: I asked a similar question and was told the scientists wanted it to exact match.

**Review by migbro — APPROVED** [review]
Thanks!

File: scripts/hotspot.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmiller15) [reply scripts/hotspot.py:139]: Well I'll definitely use that in the future as that's pretty much exactly what I made here except manually and probably much less efficiently.

**Pull request merged**